src/lib/containers/QueryCount.tsx

```diff
@@ -14,82 +14,71 @@ export type QueryCountProps = {
   token?: string
 }
 
-type QueryCountState = {
-  storedSqlQueryCount: {}
+const QueryCount: React.FunctionComponent<QueryCountProps> = ({
+  sql,
+  selectedFacets,
+  parens,
+  name,
+  token,
+}) => {
+  const [storedSqlQueryCount, setStoredSqlQueryCount] = useState<{}>({})
   // maps sql string to true/false, true if already made a request for this sql's query count
   // false or undefined if not
-  isCalculatingQueryCountForSql: {}
-}
+  const [isCalculatingQueryCountForSql, setIsCalculatingQueryCountForSql] = useState<{}>({})
+  let mounted = true
 
-/**
- * QueryCount shows a query count
- *
- * @export
- * @class QueryCount
- * @extends {React.Component<QueryCountProps, QueryCountState>}
- */
-export default class QueryCount extends React.Component<
-  QueryCountProps,
-  QueryCountState
-> {
-  constructor(props: QueryCountProps) {
-    super(props)
-    this.state = {
-      storedSqlQueryCount: {},
-      isCalculatingQueryCountForSql: {},
+  useEffect(() => {
+    const calculateRowCount = () => {
+      if (mounted) {
+        const entityId = parseEntityIdFromSqlStatement(sql)
+        if (
+          isCalculatingQueryCountForSql[`${sql}-${token}`] ||
+          storedSqlQueryCount[`${sql}-${token}`]
+        ) {
+          // its either in progress or its already been calculated
+          return
+        }
+        const request: QueryBundleRequest = {
+          concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+          query: {
+            sql,
+            selectedFacets,
+          },
+          entityId,
+          partMask: SynapseConstants.BUNDLE_MASK_QUERY_COUNT,
+        }
+        const newIsCalculatingQueryCountForSql = {
+          ...isCalculatingQueryCountForSql,  
+        }
+        newIsCalculatingQueryCountForSql[`${sql}-${token}`] = true
+        setIsCalculatingQueryCountForSql(newIsCalculatingQueryCountForSql)
+        debugger
+        SynapseClient.getQueryTableResults(request, token).then(data => {
+          debugger
```

Comment:
Same here, we should remove the debugger statement.

Reply:
Also removed.